vowpalwabbit/parse_args.cc

```diff
@@ -824,7 +825,9 @@ void parse_feature_tweaks(options_i& options, vw& all, std::vector<std::string>&
       all.interactions.clear();
     }
 
-    all.interactions = expanded_interactions;
+    for (auto &i:expanded_interactions){
+      all.interactions.emplace_back(i.begin(), i.end());
```

Comment:
Its really strange that half this code is treating an interaction as a string and the other half as a vector of uint8s. @jackgerrits would it make sense to covert the underlying functions in interactions.h/cc to work on `vector<vector<uint8_t>>` instead of `vector<string>`?

I know `expand_namespaces_with_recursion()` is looking for the `:` character, but AFAIK thats the only actual character comparison we do

Reply:
Also, below, `all.pairs` and `all.triples` are strings, not uint8s. Do we care about that?

Reply:
Yes, I think ideally all usages of namespaces should be consistent in their treating them as vectors of bytes

Reply:
@peterychang thanks for pointing it out, I will change that.